vcf/parser.py

```diff
@@ -83,7 +83,7 @@ def __init__(self):
         super(_vcf_metadata_parser, self).__init__()
         self.info_pattern = re.compile(r'''\#\#INFO=<
             ID=(?P<id>[^,]+),\s*
-            Number=(?P<number>-?\d+|\.|[AGR]),\s*
+            Number=(?P<number>(?:(-?\d+|\.|[AGR]))?),\s*
```

Comment:
I think it would be enough to just append a `?` to the matching group, like this:

    Number=(?P<number>-?\d+|\.|[AGR])?,\s*

This should make the match group value `None` for empty values, obsoleting the extra check in `vcf_field_count` below.